milli/src/update/index_documents/enrich.rs

```diff
@@ -49,22 +53,34 @@ pub fn enrich_documents_batch<R: Read + Seek>(
                         primary_key: primary_key.to_string(),
                         document: obkv_to_object(&first_document, &documents_batch_index)?,
                     })),
-                    None => Ok(Err(UserError::MissingPrimaryKey)),
+                    None => unreachable!("Called with reader.is_empty()"),
                 };
             }
         },
         None => {
-            let guessed = documents_batch_index
+            let mut guesses: Vec<(u16, &str)> = documents_batch_index
                 .iter()
-                .filter(|(_, name)| name.to_lowercase().contains(DEFAULT_PRIMARY_KEY))
-                .min_by_key(|(fid, _)| *fid);
-            match guessed {
-                Some((id, name)) => PrimaryKey::flat(name.as_str(), *id),
-                None if autogenerate_docids => PrimaryKey::flat(
+                .filter(|(_, name)| name.to_lowercase().ends_with(DEFAULT_PRIMARY_KEY))
+                .map(|(field_id, name)| (*field_id, name.as_str()))
+                .collect();
+
+            guesses.sort_by_key(|(field_id, _)| *field_id);
```

Comment:
Do you think it's better to sort by `field_id` or by string directly?

I feel like most of the time the most important key is the smallest one (like `id`) instead of the first appearing in the document what do you think?

Reply:
I don't have a strong opinion one way or another. I chose the order of appearance so that users can find the options easily in the JSON even if the document is huge since the options will tend to be by the top, but even this is a heuristic and I don't mind changing for the lexicographic sort.

Reply:
The logic might be a bit more complicated though because we will have to sort first by length, then lexicographically, to ensure a consistent order.

Reply:
We could sort lexicographically by the end :joy:  
```rust
guesses.sort_by(|left, right| left.chars().rev().partial_cmp(right.chars().rev()).unwrap_or(Ordering::Equal));
```
https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.partial_cmp

Honestly, it's as you wish, it'll be slower for sure, but I don't think that's a big deal personally since we were ready to run an entire indexing process.

Reply:
Changed the sort to:

```rust
            // sort the keys in a deterministic, obvious way, so that fields are always in the same order.
            guesses.sort_by(|(_, left_name), (_, right_name)| {
                // shortest name first
                left_name.len().cmp(&right_name.len()).then_with(
                    // then alphabetical order
                    || left_name.cmp(right_name),
                )
            });
```

Tell me what you think

Reply:
yep, imo, that’s perfect 👌